doc: Update Fast Pair guide to match the integration template (#13972)
nrfconnect/sdk-nrf: doc/nrf/external_comp/bt_fast_pair.rst

   The implementation passes tests of `Fast Pair Validator app`_ (beta version).
   The procedure triggered in Android settings is successful (tested with Android 11).

Integration prerequisites
*************************

> Before you start the |NCS| integration with Fast Pair, make sure that the following prerequisites are completed:

Reviewer: ```suggestion
Before you start the |NCS| integration with Fast Pair, make sure that the following prerequisites are fulfilled:
```

> * :ref:`Install the nRF Connect SDK <installation>`.

Reviewer 2: I would consider adding one more step:
```
* Install requirements of :ref:`bt_fast_pair_provision_script`. 
```
The script is automatically called during build, so it would be good to ensure that user installs all of the dependencies before build

Author: updated as : _* Install the requirements mentioned in the :ref:`bt_fast_pair_provision_script` to generate the hexadecimal file required for the :ref:`Google Fast Pair integration in the nRF Connect SDK <ug_bt_fast_pair>`._

> * Setting up a supported :term:`Development Kit (DK)`.

Reviewer: ```suggestion
* Set up a supported :term:`Development Kit (DK)`.
```

   The implementation passes tests of `Fast Pair Validator app`_ (beta version).
   The procedure triggered in Android settings is successful (tested with Android 11).

Integration prerequisites
*************************

Before you start the |NCS| integration with Fast Pair, make sure that the following prerequisites are completed:

* :ref:`Install the nRF Connect SDK <installation>`.

> * Setup of the required :term:`Development Kit (DK)`.

Reviewer 3: ```suggestion
* Setup of the supported :term:`Development Kit (DK)`.
```
tbc

Author: I have made this a bit clear. It is about setting up of DK (mainly steps mentioned in the GSG of a DK).

* During the early stages of development, you can use the debug Model ID and Anti-Spoofing Public/Private Key pair obtained by Nordic Semiconductor for local tests.
  Later on, you need to register your own Fast Pair Provider device with Google.
  The :ref:`ug_bt_fast_pair_provisioning_register` section in this document explains how to register the device and obtain the Model ID and Anti-Spoofing Public/Private Key pair.

Solution architecture
*********************

The |NCS| integrates the Fast Pair Provider role, facilitating communication between the Fast Pair Seeker (typically a smartphone) and the Provider (your device).

> The integration involves setting up Bluetooth LE advertising, interacting with the GATT service, and managing device provisioning and factory resets.

Reviewer 3: Feels like this sentence is a bit of a repetition of the "Integration overview"

Author: I had removed the "Integration Overview" section and moved a sentence to the "Solution architecture" section.
We could add the diagram demonstrating the module architecture in the "Solution architecture" section, as this is what the section is for:  https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/templates/Integration_template_README.html#solution-architecture

Reviewer 4: Yes, but my comment was more about the duplication of information from the `Integration steps` section that is not removed. The `Integration steps` section outlines the steps necessary for integration. These steps are also mentioned in this sentence and there is a chance that these two sources of the same information will get misaligned. Have you considered changing this sentence to the following?

```suggestion
The integration involves following the instructions outlined in the ref:`ug_integrating_fast_pair` section.
```

Currently, you do not list all the activities necessary in the integration steps sections such as:
- Provisioning the device
- Performing prerequisite operations 
- Tailoring protocol for specific use case

Author: okay, I misunderstood your point and I get it now. I have updated the sentence as above. Thanks.

These requirements typically help to improve user experience or security properties for the chosen use case.
To learn more, see the official `Fast Pair`_ documentation.

Applications and samples
************************

The following application and sample use the Fast Pair integration in the |NCS|:

* :ref:`nrf_desktop` application
* :ref:`fast_pair_input_device` sample

Library support
***************

The following |NCS| libraries support the Fast Pair integration:

* :ref:`bt_fast_pair_readme` library is used to implement the Bluetooth LE GATT Service required for :ref:`ug_bt_fast_pair` with the |NCS|.
* :ref:`bt_le_adv_prov_readme` generates the Fast Pair provider API.

Required scripts
****************

The :ref:`bt_fast_pair_provision_script` is required to generate the provisioning data for the device.
This script is a part of the build system of the |NCS| and is automatically invoked when the :kconfig:option:`CONFIG_BT_FAST_PAIR` Kconfig option is enabled.

Terms and licensing
*******************

The use of Google Fast Pair may be subject to Google's terms and licensing.

> Nordic Semiconductor customers should refer to the official Fast Pair documentation for development-related licensing information.

Reviewer: ```suggestion
Refer to the official Fast Pair documentation for development-related licensing information.
```
Should we have a link here?

Author: Updated with the link.

Dependencies
************

The following are the required dependencies for the Fast Pair integration:

> * :ref:`zephyr:bluetooth`

Reviewer 2: I think there is no need to go in more details here. I would leave it as is

Different use cases may require implementation of additional guidelines for your accessory firmware or specific configuration of your device model in the Google Nearby Devices console.
These requirements typically help to improve user experience or security properties for the chosen use case.
To learn more, see the official `Fast Pair`_ documentation.

Applications and samples
************************

> The following application and sample use the Fast Pair integration in |NCS|:

Reviewer 3: ```suggestion
The following application and sample use the Fast Pair integration in the |NCS|:
```

* :ref:`nrf_desktop` application
* :ref:`fast_pair_input_device` sample

Library support
***************

The following |NCS| libraries support the Fast Pair integration:

* :ref:`bt_fast_pair_readme` library is used to implement the Bluetooth® Low Energy (LE) GATT Service required for :ref:`ug_bt_fast_pair` with the |NCS|.

> * :ref:`bt_le_adv_prov_readme` provides the Fast Pair provider API.

Reviewer 3: ```suggestion
* :ref:`bt_le_adv_prov_readme` delivers the Fast Pair provider API.
```
not sure if that's the right substitution but just wanted to minimize providers

Author: May be we could use _generates_?

Reviewer 3: sounds great!

Required scripts
****************

The :ref:`bt_fast_pair_provision_script` is required to generate the provisioning data for the device.

> This script is part of the build system of |NCS| and is automatically invoked when the :kconfig:option:`CONFIG_BT_FAST_PAIR` option is enabled.

Reviewer 3: ```suggestion
This script is a part of the build system of the |NCS| and is automatically invoked when the :kconfig:option:`CONFIG_BT_FAST_PAIR` Kconfig option is enabled.
```

The following |NCS| libraries support the Fast Pair integration:

> * :ref:`bt_fast_pair_readme` library implements the Fast Pair subsystem and provides the APIs required for :ref:`ug_bt_fast_pair` with the |NCS|.

Reviewer 2: ```suggestion
* :ref:`bt_fast_pair_readme` library implements the Fast Pair GATT Service and provides the APIs required for :ref:`ug_bt_fast_pair` with the |NCS|.
```

> * :ref:`bt_le_adv_prov_readme` generates the Fast Pair advertising data and is optional for :ref:`ug_bt_fast_pair` with the |NCS|.

Reviewer 2: ```suggestion
* Google Fast Pair advertising data provider (:kconfig:option:`CONFIG_BT_ADV_PROV_FAST_PAIR`) can be used to integrate Fast Pair advertising payload to :ref:`bt_le_adv_prov_readme`.
  The Bluetooth LE advertising providers subsystem can be used to manage advertising and scan response data.
```
